### PR TITLE
Feature - 2637 - Admin Dashboard Page

### LINKED ISSUE #2637: Dashboard

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2637 

## Summary

This PR adds content to the admin dashboard page including:

- Current user name
- Buttons to main pages
- Latest requests table showing last 10

## Changes

Some changes were made to the Table component in this PR.

- Reorganized component to be in its own directory
- Added a `latestPoolCandidateSearchRequests` query to get 10 latest requests
- Added `labelledBy` prop for a11y
- Styled table based on the [prototype](https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3?node-id=301%3A1297&scaling=scale-down&page-id=0%3A1&starting-point-node-id=410%3A1984&show-proto-sidebar=1)

## Screenshot

Screen Shot 2022-05-10 at 2 44 17 PM

### CODE REVIEW

File: frontend/admin/src/js/components/dashboard/LatestRequestsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+export interface LatestRequestsTableProps {
+  data?: LatestRequestsQuery;
+}
+
+const LatestRequestsTable: React.FC<LatestRequestsTableProps> = ({ data }) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+  const paths = useAdminRoutes();
+
+  console.log(data);
```
Comment (vd1992) [inline frontend/admin/src/js/components/dashboard/LatestRequestsTable.tsx:65]: Left over console.log

File: frontend/common/src/components/Link/IconLink.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { Style } from "util";
```
Comment (vd1992) [inline frontend/common/src/components/Link/IconLink.tsx:2]: this is unused, intentional?

**Comment by vd1992** [comment]
Pool Name does not appear to sort correctly
image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Pretty great as usual Eric!
Just a few small things noted in the other comments, otherwise

- looks right
- links point to the right places

I think I only have 10 search requests stored locally so I can't verify the fetch recent 10 is working correctly

File: frontend/admin/src/js/components/dashboard/LatestRequestsTable.tsx (hunk 1 above)
Comment (esizer) [reply frontend/admin/src/js/components/dashboard/LatestRequestsTable.tsx]: Great catch, resolved with [remove stray console.log](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2710/commits/b04b14b268a071e9ff465f6e39348726c9ae7428)

File: frontend/common/src/components/Link/IconLink.tsx (hunk 2 above)
Comment (esizer) [reply frontend/common/src/components/Link/IconLink.tsx]: Not sure how that got there 😕 .

Resolved with [remove unused Style import](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2710/commits/2b2ab3c0dd3744d49ebc3e6b6b2bbb2e8967ceff)

**Comment by esizer** [comment]
> Pool Name does not appear to sort correctly

So, I took a look at this and I think it has something to do with the line in their [documentation](https://react-table.tanstack.com/docs/api/useTable):

- The data returned by an accessor should be primitive and sortable.

I used the other tables as an example and it looks like we are doing a little too much in our `accessor` key.

## Previous

We were using the accessor to render the field.

```tsx
...
accessor: ({ poolCandidateFilter }) =>
        poolCandidateFilter.pools?.map(
          (pool) =>
            pool && (
              <a key={pool.id} href={paths.poolCandidateTable(pool.id)}>
                {pool.name?.[locale]}
              </a>
            ),
        ),
...
```

## New

It's not very elegant, but we are now returning a single string from `accessor` and using the `Cell` to render the link.

```tsx
...
 accessor: ({ poolCandidateFilter }) =>
        poolCandidateFilter?.pools
          ? poolCandidateFilter.pools[0]?.name?.[locale]
          : null,
      Cell: ({ row: { original } }: { row: IRow }) =>
        original.poolCandidateFilter.pools?.map(
          (pool) =>
            pool && (
              <a key={pool.id} href={paths.poolCandidateTable(pool.id)}>
                {pool.name?.[locale]}
              </a>
            ),
        ),
...
```

Resolved by [fix pool name column sorting](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2710/commits/c5c8a1ddbbaaedcab6b0fb43824274a095ff459e)

**Pull request merged**